rmgpy/molecule/group.py

```diff
@@ -1117,18 +1117,25 @@ def updateFingerprint(self):
         isomorphism checks.
         """
         cython.declare(atom=GroupAtom, atomType=AtomType)
-        cython.declare(carbon=AtomType, nitrogen=AtomType, oxygen=AtomType, sulfur=AtomType)
-        cython.declare(isCarbon=cython.bint, isNitrogen=cython.bint, isOxygen=cython.bint, isSulfur=cython.bint, radical=cython.int)
-        
+        cython.declare(carbon=AtomType, nitrogen=AtomType, oxygen=AtomType, sulfur=AtomType, chlorine=AtomType, silicon=AtomType)
+        cython.declare(isCarbon=cython.bint, isNitrogen=cython.bint, isOxygen=cython.bint, isSulfur=cython.bint,
+                       isChlorine=cython.bint, isSilicon=cython.bint, radical=cython.int)
+
         carbon   = atomTypes['C']
         nitrogen = atomTypes['N']
         oxygen   = atomTypes['O']
+        # flourine = atomTypes['F']
         sulfur   = atomTypes['S']
+        chlorine = atomTypes['Cl']
+        silicon  = atomTypes['Si']
+        # iodine   = atomTypes['I']
         
         self.carbonCount   = 0
         self.nitrogenCount = 0
         self.oxygenCount   = 0
         self.sulfurCount   = 0
+        self.chlorineCount = 0
+        self.siliconCount = 0
         self.radicalCount  = 0
         for atom in self.vertices:
             if len(atom.atomType) == 1:
```

Comment:
This is outside the scope of your commit, but what should happen if len(atom.atomType) != 1?

Reply:
My understanding is that since it is a Group and not a Molecule, an "atom" can be defined as `1 [C,N,O] ux px cx {2,S}`, in which case we can't fingerprint it and we correctly skip this condition. I think this also means we're missing out on cases like `1 [Cs,Cd] ux px cx {2,S}` where we could fingerprint, but don't (for that atom in the Group). However, this should be working smoothly for Molecules.

Reply:
That makes sense. I wonder if there is a way to catch cases like `1 [Cs,Cd] ux px cx {2,S}` and if that would be useful. If so this is something for another PR though.